src/coreclr/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.cs

```diff
@@ -35,6 +44,7 @@ private static string MarshalToString(IntPtr arg, string argName)
         /// <param name="delegateTypeNative">Assembly qualified delegate type name</param>
         /// <param name="reserved">Extensibility parameter (currently unused)</param>
         /// <param name="functionHandle">Pointer where to store the function pointer result</param>
+        [RequiresUnreferencedCode("Native hosting is not trim compatible", Url = "https://aka.ms/dotnet-illink/nativehost")]
```

Comment:
Might be worth adding a comment that this is actually never reported to the user directly. Instead we rely on GetFunctionPointer to report warning coming from GetIsolatedComponentLoadContext.